src/Serialization/ConstructingMessageSerializer.php

```diff
@@ -49,7 +41,7 @@ public function serializeMessage(Message $message): array
         ];
     }
 
-    public function unserializePayload(array $payload): Generator
+    public function unserializePayload(array $payload): Message
```

Comment:
That's correct, it could merge two events into one or split one into multiple. I've removed that because it made other routines more complex than needed.